clang/include/clang/Basic/AttrDocs.td

```diff
@@ -1825,6 +1825,26 @@ be applied multiple times to the same loop.
   }];
 }
 
+def SYCLDeviceIndirectlyCallableDocs : Documentation {
+  let Category = DocCatFunction;
+  let Heading = "intel::device_indirectly_callable";
+  let Content = [{
+This attribute can only be applied to a functions and indicates that the
+function must be treated as device one and must be emitted even if it has no
+direct uses from other SYCL device functions. However, it cannot be applied to
+functions marked as 'static', functions declared within an anonymous
+space or class member functions.
+
+It also means that function should be available externally and
+cannot be optimized out due to reachability analysis or by any other
+optimization.
+
+This attribute allows to pass name and address of this function to special
```

Comment:
Perhaps the following phrasing is a little easier to read:  "This attribute allows users to pass the name and address of the function to a special ``cl::sycl::intel::get_device_func_ptr`` API call which extracts the device function pointer for the specified function, ..." 

Although I don't quite understand the last few words in this context ", SYCL program and SYCL device." So perhaps the rephrasing isn't completely accurate. Does it mean it will extract the device function pointer on both the SYCL host and device?

Reply:
I think I will remove the last part ", SYCL program and SYCL device." since it is related to the description of the mentioned API call and not to the attribute itself.